### PR TITLE
763 sign docker image

### PR DESCRIPTION
**Pull request by ggarri**
## Fixed Issue(s)
#763

## Changelog

- [ ] I thought about the changelog and included a [changelog update if required](https://github.com/consensys/quorum-key-manager/blob/master/CHANGELOG.md).

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -12,11 +12,23 @@ commands:
...
+            echo "Login in $DOCKER_REGISTRY with user $DOCKER_USER"
             echo $DOCKER_PASS | docker login -u $DOCKER_USER --password-stdin $DOCKER_REGISTRY
-            echo $DOCKER_PASS_2 | docker login -u $DOCKER_USER_2 --password-stdin $DOCKER_REGISTRY_2
+            echo "Login in $DOCKER_REGISTRY_2 with user $DOCKER_USER_RW"
+            echo $DOCKER_PASSWORD_RW | docker login -u $DOCKER_USER_RW --password-stdin $DOCKER_REGISTRY_2
+      - run:
+          name: DCT signing setup
+          command: |
+            mkdir -p $HOME/.docker/trust/private
+            echo $DCT_KEY | base64 --decode > $HOME/.docker/trust/private/$DCT_HASH.key
```
Comment (6l20) [inline .circleci/config.yml:29]: Why is the key named DCT_HASH once decoded ?

File: .circleci/config.yml [hunk 2]
```diff
@@ -218,12 +202,57 @@ jobs:
...
+              if [ $TARGETOS != "linux" ]; then
+                echo "Targeted OS not supported $TARGETOS"
+                exit 1
+              fi
+
+              if [ $TARGETARCH == "amd64" ]; then
+                DOCKER_IMAGE_TAG="${DOCKER_IMAGE_REF}"
+              else
+                DOCKER_IMAGE_TAG="${DOCKER_IMAGE_REF}.${TARGETARCH}"
+              fi
```
Comment (6l20) [inline .circleci/config.yml:228]: We are loosing the `amd64` tag of the image, maybe we should introduce a new convention here with explicitly tagged arch ?

File: .circleci/config.yml (hunk 1 above)
Comment (ggarri) [reply .circleci/config.yml:29]: Those values are being defined in the `dockerhub-opsquorum-dct` CircleCI context by Joshua, I guess, for the opsquorum team. So it is nothing I can do here to rename it

File: .circleci/config.yml (hunk 2 above)
Comment (ggarri) [reply .circleci/config.yml:228]: I decided to follow similar strategy than `mysql-server` where the amd64 is consider the default and the arm64 is tagged specifying the arch, although I did a different naming convention which I would be open to discuss if you propose something else

image

https://hub.docker.com/r/mysql/mysql-server/tags?page=1

**Review by 6l20 — APPROVED** [review]

**Pull request merged**